core/commands/files/files.go

```diff
@@ -0,0 +1,555 @@
+package commands
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	gopath "path"
+	"strings"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+	core "github.com/ipfs/go-ipfs/core"
+	dag "github.com/ipfs/go-ipfs/merkledag"
+	mfs "github.com/ipfs/go-ipfs/mfs"
+	path "github.com/ipfs/go-ipfs/path"
+	ft "github.com/ipfs/go-ipfs/unixfs"
+	u "github.com/ipfs/go-ipfs/util"
+)
+
+var log = u.Logger("cmds/files")
+
+var FilesCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Manipulate unixfs files",
+		ShortDescription: `
+Files is an API for manipulating ipfs objects as if they were a unix filesystem.
+`,
+	},
+	Subcommands: map[string]*cmds.Command{
+		"read":  FilesReadCmd,
+		"write": FilesWriteCmd,
+		"mv":    FilesMvCmd,
+		"cp":    FilesCpCmd,
+		"ls":    FilesLsCmd,
+		"mkdir": FilesMkdirCmd,
+		"stat":  FilesStatCmd,
+		"rm":    FilesRmCmd,
+	},
+}
+
+var FilesStatCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "display file status",
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("path", true, false, "path to node to stat"),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		node, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		path := req.Arguments()[0]
+		fsn, err := mfs.Lookup(node.FilesRoot, path)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		nd, err := fsn.GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		k, err := nd.Key()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		res.SetOutput(&Object{
+			Hash: k.B58String(),
```

Comment:
probably want more here, maybe:
- size
- cumulative size
- # of files (if dir)